### PR TITLE
Finish alignment of RHEL8 OSPP profile with Common Criteria

### PR DESCRIPTION
**Pull request by jan-cerny**
Adds missing rules or removes unnecessary rules so that OSPP is on par with RHEL 8 CC.
The temporary `ccc` profile is removed.
I have noticed that some of the recently introduced CCEs are duplicated, so I have fixed them by assigning new CCEs.

The rules that remain to be implemented are mentioned as TODO in comments.

### CODE REVIEW

**Comment by jan-cerny** [comment]
it's ready for a review

File: rhel8/profiles/ospp.profile [hunk 1]
```diff
@@ -8,848 +8,534 @@ description: |-
...
 
-    ## RHEL 8 CCE-80869-1: Ensure SELinux is Enforcing
-    - var_selinux_state=enforcing
-    - selinux_state
+    ## RHEL 8 CCE-80855-0: Require Authentication for Single User Mode
+    - require_singleuser_auth
 
-    ## RHEL 8 CCE-80868-3: Configure SELinux Policy
-    - var_selinux_policy_name=targeted
-    - selinux_policytype
```
Comment (shawndwells) [inline rhel8/profiles/ospp.profile]: NACK. Looks like several rules were removed, such as SELinux. Need to refactor and add them all back in.

Comment (jan-cerny) [reply rhel8/profiles/ospp.profile]: but SELinux rules shouldn't be there

Comment (jan-cerny) [reply rhel8/profiles/ospp.profile]: and SSHD ciphers and MACs should be solved by crypto-policies in future

Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: It's a required part of our OSPP profile.

Comment (comps) [reply rhel8/profiles/ospp.profile]: @shawndwells Can you please provide any references? ... SELinux is not part of NIAP's OSPPv4.2.1 nor is it implied in our ST. Not having it in the profile doesn't mean it's disabled, just that it's not checked/remediated.

Comment (shawndwells) [reply rhel8/profiles/ospp.profile]: OSPP 4.2 can be found here:
https://www.niap-ccevs.org/MMO/PP/-442-/#SecurityObjectivesTOE

With the config annex:
https://www.niap-ccevs.org/MMO/PP/-442ConfigAnnex-/

NIAP made it clear they will reject any certification package lacking SELinux. And I've made it clear I will reject any engineering work lacking SELinux as well.

Comment (yuumasato) [reply rhel8/profiles/ospp.profile]: @jan-cerny Can you add the selinux rules back? So we can get this merged?

Comment (jan-cerny) [reply rhel8/profiles/ospp.profile]: @yuumasato yes, I'm adding them back right now.

Comment (jan-cerny) [reply rhel8/profiles/ospp.profile]: @yuumasato @shawndwells I have re-added  SELinux rules in 98160f6

File: rhel8/profiles/ospp.profile [hunk 2]
```diff
@@ -887,26 +573,73 @@ selections:
...
+    # - var_system_crypto_policy=<POLICY>
+
+
+    #################################################################
+    ## SSH Client
+    #################################################################
+
+    # TODO: HostbasedAuthentication no
+
+    # TODO: GSSAPIAuthentication no
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply rhel8/profiles/ospp.profile]: This is for SSH client. That one was for SSH server. The rule for SSH server is still there (on line 569).

**Comment by yuumasato** [comment]
>  What rules were added/removed?

The list of rules added are: 
```
- accounts_password_pam_unix_remember
- accounts_passwords_pam_faillock_deny
- accounts_passwords_pam_faillock_interval
- accounts_passwords_pam_faillock_unlock_time
- audit_rules_etc_group_open
- audit_rules_etc_group_open_by_handle_at
- audit_rules_etc_group_openat
- audit_rules_etc_gshadow_open
- audit_rules_etc_gshadow_open_by_handle_at
- audit_rules_etc_gshadow_openat
- audit_rules_etc_passwd_open
- audit_rules_etc_passwd_open_by_handle_at
- audit_rules_etc_passwd_openat
- audit_rules_etc_shadow_open
- audit_rules_etc_shadow_open_by_handle_at
- audit_rules_etc_shadow_openat
- audit_rules_execution_seunshare
- audit_rules_kernel_module_loading_delete
- audit_rules_kernel_module_loading_finit
- audit_rules_kernel_module_loading_init
- auditd_audispd_syslog_plugin_activated
- auditd_data_retention_flush
- chronyd_client_only
- dconf_gnome_banner_enabled
- dconf_gnome_login_banner_text
- enable_dracut_fips_module
- ensure_gpgcheck_globally_activated
- ensure_gpgcheck_local_packages
- ensure_gpgcheck_never_disabled
- grub2_audit_argument
- grub2_audit_backlog_limit_argument
- grub2_disable_interactive_boot
- grub2_page_poison_argument
- grub2_slub_debug_argument
- grub2_uefi_password
- grub2_vsyscall_argument
- mount_option_boot_nodev
- mount_option_boot_nosuid
- mount_option_dev_shm_nodev
- mount_option_dev_shm_noexec
- mount_option_dev_shm_nosuid
- mount_option_home_nodev
- mount_option_tmp_nodev
- mount_option_tmp_noexec
- mount_option_tmp_nosuid
- mount_option_var_log_audit_nodev
- mount_option_var_log_audit_noexec
- mount_option_var_log_audit_nosuid
- mount_option_var_log_nodev
- mount_option_var_log_noexec
- mount_option_var_log_nosuid
- mount_option_var_nodev
- mount_option_var_tmp_noexec
- mount_option_var_tmp_nosuid
- package_tmux_installed
- require_singleuser_auth
- service_auditd_enabled
- service_debug-shell_disabled
- sssd_run_as_sssd_user
- sysctl_kernel_perf_event_paranoid
- sysctl_kernel_unprivileged_bpf_disabled
- sysctl_net_core_bpf_jit_harden
- var_auditd_flush=incremental_async
- sshd_idle_timeout_value=10_minutes
- var_sshd_set_keepalive=0
- var_accounts_passwords_pam_faillock_deny=3
- var_accounts_passwords_pam_faillock_fail_interval=900
- var_accounts_passwords_pam_faillock_unlock_time=never
- login_banner_text=usgcb_default
- var_accounts_max_concurrent_login_sessions=10
- var_password_pam_unix_remember=5
- var_accounts_password_minlen_login_defs=12
- var_password_pam_minlen=12
- var_password_pam_ocredit=1
- var_password_pam_dcredit=1
- var_password_pam_ucredit=1
- var_password_pam_lcredit=1
- var_password_pam_difok=4
- var_password_pam_maxrepeat=3
- var_password_pam_maxclassrepeat=4
- var_accounts_user_umask=027
- var_selinux_state=enforcing
- var_selinux_policy_name=targeted
```
Generated by `utils/profile_tool.py`

@jan-cerny Did you remove any rules?


> Could you re-order the rules to align with the ospp kickstart?

@shawndwells Do you feel strongly about this?
@jan-cerny What do you think?

**Comment by shawndwells** [comment]
> > Could you re-order the rules to align with the ospp kickstart?
> 
> @shawndwells Do you feel strongly about this?

If we ship an ospp-kickstart through Steve Grubb's work, then yes, the rule order should be aligned. Need to make it easier for users to compare the two and ensure everything in the kickstart is presented in the profile. Scrolling all around through the two documents will be incredibly frustrating.

If the ospp-kickstart will instead be incorporated completely into ComplianceAsCode, then there is no need to worry about it. There will only be one source of truth (the ComplianceAsCode profile).

**Comment by shawndwells** [comment]
> > What rules were added/removed?
> 
> The list of rules added are:

Looks like a subset of what was added in https://github.com/ComplianceAsCode/content/pull/4586. Suggest merging https://github.com/ComplianceAsCode/content/pull/4586.

**Comment by yuumasato** [comment]
> If the ospp-kickstart will instead be incorporated completely into ComplianceAsCode, then there is no need to worry about it. There will only be one source of truth (the ComplianceAsCode profile).

Yes, it will be incorporated. The goal is to have the cleanest kickstart possible.

**Comment by jan-cerny** [comment]
rebased on the top of master

**Pull request closed without merging**